### PR TITLE
12 Implemented filter section component

### PR DESCRIPTION
**Pull request by ishootdice**

### CODE REVIEW

**Comment by Anatolii-Ivanov** [comment]
Can you add a screenshot?

**Comment by ishootdice** [comment]
Yeah, of course
image

File: frontend/src/app/components/filter-section/filter-section.component.html [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<div class="filters-section-container">
+    <div class="filter-options-wrp">
+        <div class="filter-options-btn-wrp">
+            <div (click)="showLevelsList()" class="filter-options-btn">
+                <div *ngIf="this.selectedLanguageFilters.size > 0" class="filters-count-group">
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.html:5]:
```suggestion
                <div *ngIf="this.selectedLanguageFilters.size" class="filters-count-group">
```

File: frontend/src/app/components/filter-section/filter-section.component.html [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                    <div class="filters-count-val">
+                        {{this.selectedLanguageFilters.size}}
+                    </div>
+                </div>
+                <div class="lang-level-subtitle">
+                    Level
+                </div>
+                <div class="lang-level-title">
+                    <span *ngIf="this.selectedLanguageFilters.size > 0">
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.html:14]:
```suggestion
                    <span *ngIf="this.selectedLanguageFilters.size">
```

File: frontend/src/app/components/filter-section/filter-section.component.html [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                        {{this.selectedLanguageFilters.keys().next().value}}
+                    </span>
+                    <span *ngIf="this.selectedLanguageFilters.size === 0">
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.html:17]:
```suggestion
                    <span *ngIf="!this.selectedLanguageFilters.size">
```

File: frontend/src/app/components/filter-section/filter-section.component.html [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                    <div class="filters-count-val">
+                        {{this.selectedLanguageFilters.size}}
+                    </div>
+                </div>
+                <div class="lang-level-subtitle">
+                    Level
+                </div>
+                <div class="lang-level-title">
+                    <span *ngIf="this.selectedLanguageFilters.size > 0">
+                        {{this.selectedLanguageFilters.keys().next().value}}
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.html:15]: We need to simplify this line

File: frontend/src/app/components/filter-section/filter-section.component.html [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+
+        <div class="filter-options-btn-wrp">
+            <div class="filter-options-btn" (click)="showTopicsList()">
+                <div *ngIf="this.selectedTopicsFilters.size > 0" class="filters-count-group">
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.html:47]: see previous comments

File: frontend/src/app/components/filter-section/filter-section.component.html [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                <div class="dropdown-icon-wrp">
+                    <mat-icon>expand_more</mat-icon>
+                </div>
+            </div>
+
+            <div *ngIf="this.showTopicsDropdown" class="filters-dropdown">
+                <div *ngFor="let topic of this.topics">
+                    <div class="topics-list-option">
+                        <div class="list-option-content">
+                            <input type="checkbox" [checked]="this.selectedTopicsFilters.has(topic.title)" (click)="selectTopic(topic.title)">
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.html:64]: Move this to the ts file and replace it in all similar places

File: frontend/src/app/components/filter-section/filter-section.component.html [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                                </div>
+                                <div class="filters-option-description">
+                                    {{level.desc}}
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.html:43]: Do we really need so many divs?

File: frontend/src/app/components/filter-section/filter-section.component.html [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    </div>
+    <div class="selected-filters-wrp">
+        <div class="selected-filters-list">
+            <div style="margin: 0px;" *ngFor="let selectedTopic of this.selectedLanguageFilters">
+                <div *ngIf="selectedTopic" class="selected-filter-item">
+                    {{selectedTopic}}
+                    <mat-icon (click)="selectLangLevel(selectedTopic)">close</mat-icon>
+                </div>
+            </div>
+            <div style="margin: 0px;" *ngFor="let selectedTopic of this.selectedTopicsFilters">
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.html:84]: Please do not use inline styles

File: frontend/src/app/components/filter-section/filter-section.component.sass [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+    justify-content: space-between
+
+.filter-options-wrp
+    width: 400px
+    height: 100%
+    display: flex
+    justify-content: space-between    
+
+.filter-options-btn-wrp
+    width: 195px
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.sass:16]: In would be better do not use px

File: frontend/src/app/components/filter-section/filter-section.component.sass [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+    .lang-level-title
+        position: absolute
+        top: 17px
+        left: 50px
+
+
+.filter-options-btn:hover
+    cursor: pointer
+       
+
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.sass:68]: please remove all extra lines

File: frontend/src/app/components/filter-section/filter-section.component.sass [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+.filters-dropdown
+    position: absolute
+    width: 195px
+    height: 228px
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.sass:72]: Please avoid using so specific px sizes

File: frontend/src/app/components/filter-section/filter-section.component.spec.ts [hunk 12]
```diff
@@ -0,0 +1,23 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { FilterSectionComponent } from './filter-section.component';
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.spec.ts:3]: You can delete this file for now

File: frontend/src/app/components/filter-section/filter-section.component.ts [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+export class FilterSectionComponent implements OnInit {
+
+  public showLangDropdown = false;
+  public showTopicsDropdown = false;
+
+  public selectedLanguageCount = 0;
+  public selectedTopicsCount = 0;
+
+  public selectedLanguageFilters = new Set();
+  public selectedTopicsFilters = new Set();
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.ts:17]: We need to create a general dropdown component and reuse it in all places where we need it

File: frontend/src/app/components/filter-section/filter-section.component.ts [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+
+  public topics = [
+    {
+      title: 'Traveling',
+    },
+    {
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.ts:23]: Please move this to .utils.ts file

File: frontend/src/app/components/filter-section/filter-section.component.ts [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+    this.showTopicsDropdown = !this.showTopicsDropdown;
+  }
+
+  selectLangLevel(title: any){
+    if(this.selectedLanguageFilters.has(title)){
+      this.selectedLanguageFilters.delete(title);
+    }
+    else{
+      this.selectedLanguageFilters.add(title);
+    }
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.ts:95]: Please avoid mutations

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/components/filter-section/filter-section.component.ts [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+  public topics = topicsSample;
+  public langLevels = langLevelsSample;
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  removeLangLevel(title: any){
```
Comment (tatianahutii) [inline frontend/src/app/components/filter-section/filter-section.component.ts:28]: Please do not use any

File: frontend/src/app/components/dropdown/dropdown.component.ts [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+    }
+  }
+
+  showDropdownMenu(){
+    this.showDropdown = !this.showDropdown;
+  }
+
+  reset(){
+
+  }
```
Comment (tatianahutii) [inline frontend/src/app/components/dropdown/dropdown.component.ts:49]: Delete empty methods

File: frontend/src/app/components/dropdown/dropdown.component.ts [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+  public showDropdown = false;
+
+  public selectedItems = new Set<string>();
+
+  constructor() { }
+
+  ngOnInit(): void {
+    this.resetEvent.subscribe(() => {
+      this.selectedItems = new Set();
+    })
```
Comment (tatianahutii) [inline frontend/src/app/components/dropdown/dropdown.component.ts:27]: you need to unsibscribe

File: frontend/src/app/components/dropdown/dropdown.component.sass [hunk 19]
```diff
@@ -0,0 +1,122 @@
...
+    font-weight: 700
+    color: $main-color
+
+
+
+
+
+
+
+
```
Comment (tatianahutii) [inline frontend/src/app/components/dropdown/dropdown.component.sass:125]: Please delete extra space

File: frontend/src/app/components/dropdown/dropdown.component.sass [hunk 20]
```diff
@@ -0,0 +1,122 @@
...
+input[type="checkbox"]
+        appearance: none
+        box-sizing: border-box
+        margin: 0
+        width: 16px
+        height: 16px
+        top: 1px
+        accent-color: $main-color
+        background-color: #fff
```
Comment (tatianahutii) [inline frontend/src/app/components/dropdown/dropdown.component.sass:80]: Please use color variables

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — DISMISSED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by SidletskyiOlexandr — APPROVED** [review]

**Pull request merged**